wrangler-transform/src/main/java/io/cdap/wrangler/Wrangler.java

```diff
@@ -498,6 +500,16 @@ public static class Config extends PluginConfig {
     @Macro
     private String precondition;
 
+    @Name(NAME_PRECONDITION_SQL)
+    @Description("Precondition expression specifying filtering before applying directives (true to filter)")
+    @Macro
+    private String preconditionSql;
```

Comment:
Please update validation rules for these fields.

Precondition SQL should be not-empty when SQL mode is selected.

Precondition in Jexl should be not-empty as well when the jexl mode is selected.